slides/3-foundations-of-javascript.html

```diff
@@ -388,31 +395,113 @@ <h2><i class="fab fa-js"></i> Foundations of JavaScript</h2>
                         </textarea>
                     </section>
 
-                    <section data-markdown style="font-size: 0.7em;">
+                    <section data-markdown style="font-size: 0.65em;">
                         <textarea data-template>
                             ## <i class="fab fa-js"></i> JavaScript's Types
 
                             ECMAScript defines 7 **primitive** (Immutable) types for values:
+
+                            - `Undefined`: Unique value `undefined`
+                            - `Number`: Real or integer number (e.g. `3.14`, `42`)
+                            - `Boolean`: `true` or `false`
+                            - `String`: Character sequence, whose literals begin and end with single or double quotes (e.g. `"HEIG-VD"`, `'hello'`)
+                            - `BigIng`: Arbitrary-precision integers, whose literals end with an `n` (e.g. `9007199254740992n`)
+                            - `Symbol`: Grobally unique values usable as identifiers or keys in objects (e.g. `Symbol()`, `Symbol("description")`)
+                            - `Null`: Unique value `null`
+
+                            In a dynamic language you don't specify the type when you declare a variable and the type of a variable can change.
                             
+                            ECMAScript also defines a special mutable type called **object** for collections of properties (objects and array).
+
+                            https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Data_types
+
+                            Notes:
+
+                            The `Symbol` data type is special.
```

Comment:
Added information about `Symbol`, it's not an easy one to grasp, so it deserves some attention.